tools/config_cli.py

```
@@ -345,31 +347,46 @@ def _infer_base_from_defaults(child_path: Path, child_cfg_raw: DictConfig) -> Pa
-        base_cfg_raw = OmegaConf.load(base_path)
-        if not isinstance(base_cfg_raw, DictConfig):
-            raise TypeError(
-                f"Config at {base_path} must be a mapping (DictConfig), got {type(base_cfg_raw)}"
-            )
-        base_resolved = OmegaConf.to_container(base_cfg_raw)
+        if base_path not in base_config_cache:
+            base_cfg_raw = OmegaConf.load(base_path)
+            if not isinstance(base_cfg_raw, DictConfig):
+                raise TypeError(
+                    f"Config at {base_path} must be a mapping (DictConfig), got {type(base_cfg_raw)}"
+                )
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)